lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -147,6 +149,18 @@ async function runSmokeTest(smokeTestDefn, testOptions) {
     report = getAssertionReport(result, expectations, {isDebug});
     if (report.failed) {
       bufferedConsole.log(`  ${getAssertionLog(report.failed)} failed.`);
+
+      // For CI, save to directory to be uploaded.
```

Comment:
what do you think about putting this in `smokehouse-bin.js` so `smokehouse.js` doesn't need to know anything about the filesystem (even if always guarded by `process.env.CI`)? The result/log just needs to be added to `SmokehouseResult`.

Reply:
each failed `report` would need to be tracked, and all of them returned as part of SmokehouseResult. seems like a larger change than you'd intend to make here?

Reply:
that seems ok to me. It's just an array of stuff that's being passed back to `runSmokeTest` anyways.